### ISSUE TITLE
[Question] Is it possible to customize the actual shell command used to launch GDB?

### ISSUE THREAD

**Issue by renyhp** [microsoft/vscode-cpptools#4685]
When starting debugging, I would like to execute a shell command, _before_ running GDB, and _in the same shell_ as GDB.
Also, I would like to control the command line arguments passed to GDB.
Is either of these possible?
Thanks in advance.

**Comment by pieandcakes** [comment]
@renyhp you can add additional command line arguments by using `miDebuggerArgs` in your `launch.json`. Alternatively if you want to specify other commands before the debugger is launched, I would suggest looking at creating a shell script that passes on all the parameters that it is called with to `gdb`. this way you can call whatever other items you want prior to `gdb` launching and add additional parameters.

This is not a traditional scenario so you may need to do some debugging to figure out how to get it working.

**Comment by renyhp** [comment]
Okay, thank you. Let's just ignore the idea of passing different arguments to gdb, I have some trouble getting to execute a script. I will describe my problem here, but I understand this might not be the right place to do it; I will be grateful if anyone can point me somewhere else to post my problem or some documentation that can help me.

I am trying to debug a program whose path is contained in an environment variable, which in turn is set by a shell script. The program must also be able to access the same environment variable.
I have tried to create a shell script `mydebug.sh` with the following content:
```
. /path/to/the/shell/script.sh
/usr/bin/gdb "$@"
```
In my `launch.json` I set
```
"program": "${env:THEVARIABLE}/myprogram.exe"
"miDebuggerPath": "/path/to/mydebug.sh"
```
This doesn't work as expected: the program cannot be called.
By the way, maybe my understanding of this is flawed somewhere, because if I launch `mydebug.sh` from the shell I get the following:
```
$ mydebug.sh
[...]
(gdb) shell echo $THEVARIABLE
[correct value of the variable]
(gdb) quit
$ echo $THEVARIABLE
[not the same value anymore]
```

I also tried adding
```
{
    "description": "set env variables",
    "text": "shell source /path/to/the/shell/script.sh"
}
```
to `setupCommands`, but this still doesn't work.
